chartlets.js/packages/lib/src/actions/handleHostStoreChange.ts

```diff
@@ -43,29 +47,76 @@ export function handleHostStoreChange() {
     contributionsRecord,
     hostStore,
   );
-  if (callbackRequests && callbackRequests.length > 0) {
-    invokeCallbacks(callbackRequests);
+
+  const filteredCallbackRequests = callbackRequests.filter(
+    (callbackRequest): callbackRequest is CallbackRequest =>
+      callbackRequest !== undefined,
+  );
+  if (filteredCallbackRequests && filteredCallbackRequests.length > 0) {
+    invokeCallbacks(filteredCallbackRequests);
   }
 }
 
-function getCallbackRequests(
+// Exporting for testing only
+export function getCallbackRequests(
   propertyRefs: PropertyRef[],
   contributionsRecord: Record<string, ContributionState[]>,
   hostStore: HostStore,
-): CallbackRequest[] {
-  return propertyRefs.map((propertyRef) => {
-    const contributions = contributionsRecord[propertyRef.contribPoint];
-    const contribution = contributions[propertyRef.contribIndex];
-    const callback = contribution.callbacks![propertyRef.callbackIndex];
-    const inputValues = getInputValues(
-      callback.inputs!,
-      contribution,
+): (CallbackRequest | undefined)[] {
+  const { lastCallbackInputValues } = store.getState();
+  return propertyRefs.map((propertyRef) =>
+    getCallbackRequest(
+      propertyRef,
+      lastCallbackInputValues,
+      contributionsRecord,
       hostStore,
-    );
-    return { ...propertyRef, inputValues };
-  });
+    ),
+  );
 }
 
+const getCallbackRequest = (
+  propertyRef: PropertyRef,
+  lastCallbackInputValues: Record<string, unknown[]> | undefined,
+  contributionsRecord: Record<string, ContributionState[]>,
+  hostStore: HostStore,
+) => {
+  const contributions = contributionsRecord[propertyRef.contribPoint];
+  const contribution = contributions[propertyRef.contribIndex];
+  const callback = contribution.callbacks![propertyRef.callbackIndex];
+  const _inputValues = getInputValues(
+    callback.inputs!,
+    contribution,
+    hostStore,
+  );
+
+  // This is a dummy function created to memoize the _inputValues values
+  const _getInputValues = (_inputValues: unknown[]): unknown[] => {
+    return _inputValues;
+  };
+  const memoizedInputValues = memoize(_getInputValues);
```

Comment:
This doesn't make sense to me. Memoization of functions must be global. Otherwise there is no caching effect as the cache is created every time you create the memoized version.